extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/QuarkusRecoveryService.java

```diff
@@ -51,4 +51,11 @@ public void create() {
         }
         xaResources.clear();
     }
+
+    @Override
+    public void destroy() {
+        super.destroy();
+        isCreated = false;
+        recoveryManagerService = null;
```

Comment:
this looks good for the purposes of this specific issue, but shouldn't this `QuarkusRecoveryService` be mindful about concurrency?
I see neither the initialization/destruction being safeguarded against racy invocations, nor the overriden instance methods - are these methods protected externally?

Reply:
@zhfeng will you respond to Sanne's point about lack of protection against concurrent access, thanks.

Reply:
Well, I think the real working codes are in [RecoveryManager.java](https://github.com/jbosstm/narayana/blob/main/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/recovery/RecoveryManager.java#L173-L207). Both of initialize/terminate methods are protected by `synchronized`. So it should be safe for concurrency access.

Reply:
As far as I can tell, `RecoveryManagerService` calls `RecoveryManager`. So it's possible in theory to have `destroy` being executed concurrently to e.g. `create`, which could result in the recovery manager being in a "stopped" status while the service is in a "running" status (`isCreated == true`). Adding `synchronized` to lifecycle methods would prevent that.

That being said, the superclass `RecoveryManagerService` seems to have the same problem, so I suppose if nobody ever reported a problem in other projects... it may be safe in practice.